examples/with-nextjs-and-clerk-auth/src/components/NavigationMenu/NavigationListItem.tsx

```diff
@@ -0,0 +1,54 @@
+'use client';
+
+import { ReactNode } from 'react';
+
+import Link from 'next/link';
+import { usePathname } from 'next/navigation';
+
+import {
+  Box,
+  ListItem,
+  ListItemButton,
+  ListItemButtonProps,
+  ListItemIcon,
+  ListItemText,
+  Typography,
+} from '@mui/material';
+
+type NavigationLinkProps = {
+  children?: ReactNode;
+  endIcon?: ReactNode;
+  href?: string;
+  icon: ReactNode;
+  onClick?: ListItemButtonProps['onClick'];
+};
+
+export const NavigationListItem = ({
+  endIcon,
+  children,
+  href,
+  icon,
+  onClick,
+}: NavigationLinkProps) => {
+  const pathname = usePathname();
+  const selected = pathname === href;
+
+  const buttonProps =
+    href === undefined
+      ? { component: 'span', onClick }
```

Comment:
We need to pass a custom component only in the case if we pass the `href`. In the same time, the link can also handle `onClick` callbacks.
```ts
const buttonProps =
    href !== undefined ? { component: Link, href } : undefined;
// ..
<ListItemButton selected={selected} {...buttonProps}>
```

Reply:
Sorry for the previous incorrect example, you missed `onClick`:
```tsx
<ListItemButton selected={selected} onClick={onClick} {...buttonProps}>
```